### ISSUE TITLE
Upgrading core ("nx") from 15.1.1 to 15.2.0 breaks custom builder

### ISSUE THREAD

**Issue by woppa684** [nrwl/nx#13630]
### Current Behavior

I have a custom builder, where I wrap the existing Angular builder, looking something like:

```ts
import { JsonObject } from '@angular-devkit/core';
import { BuilderContext, BuilderOutput, createBuilder } from '@angular-devkit/architect';
import { executeWebpackBrowserBuilder, BrowserBuilderSchema } from '@nrwl/angular/executors';
import { Observable } from 'rxjs';
import { NgApplicationBuildSchema } from './schema';

export function buildExecutor(options: NgApplicationBuildSchema, context: BuilderContext): Observable<BuilderOutput> {
    return new Observable<BuilderOutput>(subscriber => {
        const internal = executeWebpackBrowserBuilder(options, context).subscribe({
            next: output => {
                subscriber.next(output);
            },
            error: error => { throw error; },
            complete: async () => {
                // I execute my custom additional code here and then complete
                subscriber.complete();
            },
        });

        return () => internal.unsubscribe();
    });
}

export default createBuilder<JsonObject & BrowserBuilderSchema>(buildExecutor);
```

All of this works fine with Nx 15.1.1 but break with 15.2.0, I then get the following error:

```
TypeError: implementation is not a function
```

Which seems correct, since "implementation" here is the result of `createBuilder`, which is an object containing 2 symbols for name and version and a handler function. What made this break in 15.2.0?

It definitely is in the update of the "nx" package from 15.1.1 to 15.2.0 because I upgraded and downgraded that single package just to be sure. I have looked through the commits but I cannot find the culprit.

### Expected Behavior

A custom builder still works, so, no error :)

### Steps to Reproduce

It's hard to reproduce with above repo since custom local builders don't work anyway (Angular architect tries to include `@<scope>/<builder>/package.json`. But the idea would be to just have a custom builder, even the smallest one already triggers the error:

```ts
import { JsonObject } from '@angular-devkit/core';
import { createBuilder } from '@angular-devkit/architect';
import { executeWebpackBrowserBuilder, BrowserBuilderSchema } from '@nrwl/angular/executors';

export default createBuilder<JsonObject & BrowserBuilderSchema>(executeWebpackBrowserBuilder);
```

Which to me is weird, since the webpack-browser builder from `@nrwl/angular` also does the exact same thing.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.18.1
   OS   : win32 x64
   pnpm : 7.17.1

   nx : 15.2.4
   @nrwl/angular : 15.2.4
   @nrwl/cypress : 15.2.4
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.2.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.2.4
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.2.4
   @nrwl/js : 15.2.4
   @nrwl/linter : 15.2.4
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : 15.2.4
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.2.4
   @nrwl/web : Not Found
   @nrwl/webpack : 15.2.4
   @nrwl/workspace : 15.2.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
         @ourscope/our-sdk
   ---------------------------------------
   Community plugins:
```

### Failure Logs

```shell
TypeError: implementation is not a function
    at C:\Repo\node_modules\.pnpm\packages\nx\src\command-line\run.ts:189:15
    at Generator.next (<anonymous>)
    at C:\Repo\node_modules\.pnpm\tslib@2.4.1\node_modules\tslib\tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (C:\Repo\node_modules\.pnpm\tslib@2.4.1\node_modules\tslib\tslib.js:114:16)
    at runExecutorInternal (C:\Repo\node_modules\.pnpm\nx@15.2.4_jb5gqfizvbzmepf2nem6vggb5e\node_modules\nx\src\command-line\run.js:79:20)
    at Generator.next (<anonymous>)
    at C:\Repo\node_modules\.pnpm\tslib@2.4.1\node_modules\tslib\tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (C:\Repo\node_modules\.pnpm\tslib@2.4.1\node_modules\tslib\tslib.js:114:16)
    at C:\Repo\node_modules\.pnpm\packages\nx\src\command-line\run.ts:290:45
    at C:\Repo\node_modules\.pnpm\packages\nx\src\utils\params.ts:78:18
    at Generator.next (<anonymous>)
    at C:\Repo\node_modules\.pnpm\tslib@2.4.1\node_modules\tslib\tslib.js:118:75
    at new Promise (<anonymous>)
```

**Comment by woppa684** [comment]
Found it! By looking at commit https://github.com/nrwl/nx/commit/8d9de7f81dc027170b2de28f0f8a012354d8695f I saw that the `executors.json` was cleaned up in a way that the builders are no longer also included in the list of executors. This was the problem, because then the builders were seen as Nx executors ( `isNgCompat` was `false` ). By removing my builders from the list of executors everything works as expected again.
